src/main/java/org/myrobotlab/service/Runtime.java

```diff
@@ -105,7 +107,7 @@
  * check for 64 bit OS and 32 bit JVM is is64bit()
  *
  */
-public class Runtime extends Service implements MessageListener, RemoteMessageHandler, Gateway, LocaleProvider {
+public class Runtime extends Service implements MessageListener, ServiceLifeCycle, RemoteMessageHandler, Gateway, LocaleProvider {

```

Comment:
Service should implement ServiceLifeCycle not Runtime to ensure all services properly implement their lifecycle methods.

Reply:
Services don't register nor release themselves .. Runtime does.
Runtime is responsible for registering, creating, starting, stopping and releasing all services.  There is no point that the service do this.

Reply:
We probably don't need an interface for ServiceLifeCycle if runtime is the only thing that implements it..